pinot-common/src/main/java/com/linkedin/pinot/common/config/SegmentPartitionConfig.java

```diff
@@ -51,9 +52,32 @@ public SegmentPartitionConfig(
    * @return Partition function for the column.
    */
   @Nullable
-  public PartitionFunction getPartitionFunction(@Nonnull String column) {
+  public String getFunctionName(@Nonnull String column) {
     ColumnPartitionConfig columnPartitionConfig = _columnPartitionMap.get(column);
-    return (columnPartitionConfig != null) ? columnPartitionConfig.getPartitionFunction() : null;
+    return (columnPartitionConfig != null) ? columnPartitionConfig.getFunctionName() : null;
+  }
+
+  /**
+   * Set the number of partitions for the specified column.
+   *
+   * @param column Column for which to set the number of partitions.
+   * @param numPartitions Number of partitions to set.
+   */
+  @JsonIgnore
+  public void setNumPartitions(String column, int numPartitions) {
```

Comment:
Is it possible for each column to have a different number of partitions? If not, can we remove this method (seems like not used as far as I could see, but I may have missed something). 
If yes, then does json package handle the two set methods correctly? Does it make sense to rename it as setNumPartitionsForColumn()?